frontend/src/schemas/SchemaList.tsx

```diff
@@ -133,57 +120,21 @@ export default function SchemaList({ schemaKind }: SchemaDisplayProps) {
       schemas.map((schema, index) => {
         const open = !!anchorEl && openMenuSchemaId === schema.id
         return (
-          <ListItem divider={index < schemas.length - 1} key={schema.id}>
-            <ListItemText>{schema.name}</ListItemText>
-            <Stack spacing={1} direction={{ xs: 'column', md: 'row' }} alignItems='center'>
-              <Chip
-                label={schema.active ? 'Active' : 'Inactive'}
-                size='small'
-                color={schema.active ? 'success' : 'warning'}
-              />
-              {schema.hidden && <Chip label='Hidden' size='small' color='error' />}
-              <Button
-                id={`schema-actions-button-${schema.id}`}
-                size='small'
-                variant='contained'
-                aria-controls={open ? `schema-actions-menu-${schema.id}` : undefined}
-                aria-haspopup='true'
-                aria-expanded={open ? 'true' : undefined}
-                endIcon={open ? <ExpandLessIcon /> : <ExpandMoreIcon />}
-                onClick={(event) => handleOpenMenu(event, schema.id)}
-              >
-                Actions
-              </Button>
-              <Menu
-                id={`schema-actions-menu-${schema.id}`}
-                open={open}
-                anchorEl={anchorEl}
-                MenuListProps={{
-                  'aria-labelledby': `schema-actions-button-${schema.id}`,
-                }}
-                anchorOrigin={{
-                  vertical: 'bottom',
-                  horizontal: 'center',
-                }}
-                transformOrigin={{
-                  vertical: 'top',
-                  horizontal: 'center',
-                }}
-                onClose={handleCloseMenu}
-              >
-                <MenuItem onClick={() => handlePatchSchema(schema.id, { active: !schema.active })}>
-                  {schema.active ? 'Mark as inactive' : 'Mark as active'}
-                </MenuItem>
-                <MenuItem onClick={() => handlePatchSchema(schema.id, { hidden: !schema.hidden })}>
-                  {schema.hidden ? 'Mark as visible' : 'Mark as hidden'}
-                </MenuItem>
-                <MenuItem onClick={() => handleDeleteSchema(schema.id)}>Delete</MenuItem>
-              </Menu>
-            </Stack>
-          </ListItem>
+          <SchemaListItem
+            key={schema.id}
+            schema={schema}
+            index={index}
+            schemasLength={schemas.length}
+            open={open}
+            anchorEl={anchorEl}
+            onMenuClose={handleCloseMenu}
+            onDeleteSchemaClick={handleDeleteSchema}
+            onOpenMenuClick={handleOpenMenu}
+            onPatchSchemaClick={handlePatchSchema}
```

Comment:
In frontend land we usually use the terminology "edit" instead of put/patch. The component doesn't need to care about the request method